### PR TITLE
added volltron development environment setup doc for vscode

### PR DESCRIPTION
**Pull request by kefeimo**
added volltron development environment setup doc for vscode

### CODE REVIEW

File: source/developing-volttron/development-environment/vscode/index.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+**Configure Path to the VOLTTRON Source:**
+- Open your VOLTTRON project folder in VSCode to automatically recognize and index the source files.
+- Your `pyproject.toml` file should be visible at the root of the project, enabling VSCode to recognize project dependencies.
+
+**Environment and Editor Settings in settings.json:**
+The `settings.json` file in VSCode is crucial for customizing the IDE environment to fit the needs of VOLTTRON development. The settings might include paths to data directories, specific Python paths, and integration with GIT features, enhancing workflow efficiency and accessibility.
+
+.. code-block:: json
+    {
+        "python.defaultInterpreterPath": "/apps/base/python3.6/bin/python",
```
Comment (schandrika) [inline source/developing-volttron/development-environment/vscode/index.rst]: does it matter if default python is pointing to python 3.6?

File: source/developing-volttron/development-environment/vscode/index.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        "terminal.integrated.env.linux": {
+            "VOLTTRON_HOME": "/home/${env:USER}/.volttron_modular"
```
Comment (schandrika) [inline source/developing-volttron/development-environment/vscode/index.rst:54]: is volttron_home the only entry here that we care about?

File: source/developing-volttron/development-environment/vscode/index.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                "program": "${file}",
+                "env": {
+                    "VOLTTRON_HOME": "~/.volttron_redo"
+                }
+            },
+            {
+                "name": "launch volttron-listener",
+                "program": "/home/user/repos/volttron-listener/src/listener/agent.py",
+                "env": {
+                    "VOLTTRON_PLATFORM_ADDRESS": "ipc://@/home/user/.volttron_redo/run/vip.socket"
```
Comment (schandrika) [inline source/developing-volttron/development-environment/vscode/index.rst]: is this all that is needed to run listener from vscode? don't you need to repeat the volttron home settings here?

File: source/developing-volttron/development-environment/vscode/index.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                }
+            }
+        ]
+    }
+
+This configuration allows developers to efficiently run and debug VOLTTRON applications using specific environment settings and utilities directly from VSCode, leveraging the powerful coding assistance and debugging tools provided by the IDE.
```
Comment (schandrika) [inline source/developing-volttron/development-environment/vscode/index.rst:122]: Is there specific setting required to setup modular volttron env? I.e. volttron-zmq, volttron-core, volttron-lib-auth ?

**Review by schandrika — CHANGES_REQUESTED** [review]

**Comment by schandrika** [comment]
could you also link to this doc from the main development doc page

File: source/developing-volttron/development-environment/vscode/index.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+==============
+VSCode can be installed directly on your Windows host and configured to connect to code within the WSL2 environment.
+
+**Installing VSCode on Windows:**
+- Download the VSCode installer for Windows from the official site and follow the installation prompts.
+- Once installed, open VSCode and install the **Remote - WSL** extension from the marketplace. This extension allows VSCode to connect and interact with Linux distributions installed via WSL2.
+
+**Configuring WSL2 with VSCode:**
+- Open VSCode.
+- Use the command palette (`Ctrl+Shift+P`) and select **Remote-WSL: New Window**. This opens a new VSCode window directly connected to your WSL2 environment.
```
Comment (schandrika) [inline source/developing-volttron/development-environment/vscode/index.rst:26]: how do you know which wsl2 container to connect to?

File: source/developing-volttron/development-environment/vscode/index.rst (hunk 1 above)
Comment (kefeimo) [reply source/developing-volttron/development-environment/vscode/index.rst]: This setting is generally not relevant especially when using the vscode python Extension "Select Python Interpreter." I can revise this part by adding more notes.

File: source/developing-volttron/development-environment/vscode/index.rst (hunk 2 above)
Comment (kefeimo) [reply source/developing-volttron/development-environment/vscode/index.rst:54]: Yes. terminal.integrated.env.linux. Other setting like "GEVENT_SUPPORT" are launch related, and should be set in launch.json

File: source/developing-volttron/development-environment/vscode/index.rst (hunk 3 above)
Comment (kefeimo) [reply source/developing-volttron/development-environment/vscode/index.rst]: Good catch, I will modify the doc

File: source/developing-volttron/development-environment/vscode/index.rst (hunk 4 above)
Comment (kefeimo) [reply source/developing-volttron/development-environment/vscode/index.rst:122]: Good point, I will add notes and provide an example settings.

File: source/developing-volttron/development-environment/vscode/index.rst (hunk 5 above)
Comment (kefeimo) [reply source/developing-volttron/development-environment/vscode/index.rst:26]: Will add more notes: For more detailed guidance and latest updates, refer to the official Visual Studio Code documentation on [Working with multiple WSL2 distributions](https://code.visualstudio.com/docs/remote/wsl#_working-with-multiple-linux-distributions).

**Comment by kefeimo** [comment]
> could you also link to this doc from the main development doc page

Done with [simplified vscode doc](https://github.com/schandrika/volttron-docs/pull/4/commits/909c1ecf8e747404cd6a1632da8335aaedae0d5e)

**Comment by schandrika** [comment]
@kefeimo for settings.json you don't have to explain all the details 1 by 1. You could add 1 settings.json with all relevant details required for typically running modular volttron - ie. if you download volttron-zmq and change say volttron-core to point to your source code - I assume such a setting would at a minimum show VOLTTRON_HOME, gevent support, python src path, may be settings to launch volttron and vctl.

**Comment by kefeimo** [comment]
> @kefeimo for settings.json you don't have to explain all the details 1 by 1. You could add 1 settings.json with all relevant details required for typically running modular volttron - ie. if you download volttron-zmq and change say volttron-core to point to your source code - I assume such a setting would at a minimum show VOLTTRON_HOME, gevent support, python src path, may be settings to launch volttron and vctl.

@schandrika  Done with [simplified vscode doc](https://github.com/schandrika/volttron-docs/pull/4/commits/909c1ecf8e747404cd6a1632da8335aaedae0d5e)

**Pull request merged**